kotlinx-coroutines-core/common/src/channels/ArrayBroadcastChannel.kt

```diff
@@ -218,13 +218,15 @@ internal class ArrayBroadcastChannel<E>(
         override val isBufferAlwaysFull: Boolean get() = error("Should not be used")
         override val isBufferFull: Boolean get() = error("Should not be used")
 
-        override fun onCancelIdempotent(wasClosed: Boolean) {
+        override fun close(cause: Throwable?): Boolean {
```

Comment:
It looks like `onCancelIdempotent` should be overridden (and do nothing) to avoid an extra bit of work.

Reply:
`ArrayBroadcastChannel.Subscriber` extends `AbstractChannel` and `onCancelIdemptotent` does not do anything there.